source/ocs/ocs.ts

```diff
@@ -53,17 +134,717 @@ export async function getUser(
   connection: OcsConnection,
   userId: string
 ) : Promise<OcsUser> {
-  let user : OcsUser;
+  let user: Promise<OcsUser>;
 
   try {
     user = await promisifiedGetUser.call(connection, userId);
   } catch (error) {
-    if (error.code !== 404) {
-      throw error;
+    user = rejectWithOcsError('Unable to find user', error.message, userId);
+  }
+
+  return user;
+}
+
+export async function setUserEnabled(
+  connection: OcsConnection,
+  userId: string,
+  isEnabled: boolean
+): Promise<boolean> {
+  let success: Promise<boolean>;
+
+  try {
+    success = await promisifiedSetUserEnabled.call(connection, userId, isEnabled);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+
+    if (error.code === 400 && error.meta && error.meta.statuscode) {
+      switch (error.meta.statuscode) {
+        case 101:
+          reason = 'user does not exist';
+          break;
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
     }
 
-    user = null;
+    success = rejectWithOcsError(`Unable to ${isEnabled ? 'enable' : 'disable'} user`, reason, userId, statusCode);
   }
 
-  return user;
+  return success;
+}
+
+export async function editUser(
+  connection: OcsConnection,
+  userId: string,
+  field: OcsEditUserField,
+  value: string
+): Promise<boolean> {
+  let userEdited: Promise<boolean>;
+
+  try {
+    userEdited = await promisifiedEditUser.call(connection, userId, field, value);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+
+    if (
+      (error.code === 400 || error.code === 401) &&
```

Comment:
I was wondering whether the entire try-catch couldn't be wrapped inside a more generic function, but stuff like this makes it hard…

Reply:
Resolving this to keep the conversation in one thread